gio/src/subclass/application.rs

```diff
@@ -266,6 +278,63 @@ pub trait ApplicationImplExt: ApplicationImpl {
             }
         }
     }
+
+    fn parent_dbus_register(
+        &self,
+        connection: &DBusConnection,
+        object_path: &str,
+    ) -> Result<(), glib::Error> {
+        unsafe {
+            let data = Self::type_data();
+            let parent_class = data.as_ref().parent_class() as *mut ffi::GApplicationClass;
+            let f = (*parent_class)
+                .dbus_register
+                .expect("No parent class implementation for \"dbus_register\"");
+            let mut err = ptr::null_mut();
+            let res = f(
+                self.obj().unsafe_cast_ref::<Application>().to_glib_none().0,
+                connection.to_glib_none().0,
+                object_path.to_glib_none().0,
+                &mut err,
+            );
+            if res == glib::ffi::GFALSE {
+                debug_assert!(!err.is_null());
+                Err(from_glib_full(err))
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    fn parent_dbus_unregister(&self, connection: &DBusConnection, object_path: &str) {
+        unsafe {
+            let data = Self::type_data();
+            let parent_class = data.as_ref().parent_class() as *mut ffi::GApplicationClass;
+            let f = (*parent_class)
+                .dbus_unregister
+                .expect("No parent class implementation for \"dbus_unregister\"");
+            f(
+                self.obj().unsafe_cast_ref::<Application>().to_glib_none().0,
+                connection.to_glib_none().0,
+                object_path.to_glib_none().0,
+            );
+        }
+    }
+
+    fn parent_name_lost(&self) -> bool {
+        unsafe {
+            let data = Self::type_data();
+            let parent_class = data.as_ref().parent_class() as *mut ffi::GApplicationClass;
+            let f = (*parent_class)
+                .name_lost
+                .expect("No parent class implementation for \"name_lost\"");
```

Comment:
Is this correct?  Most code around here uses `.expect` to access the parent function, but `handle_local_option` properly unpacks the `Option`.  I'm not sure when to use either, so I'd appreciate if you'd check this.

Reply:
Unfortunately you have to check the C code and see if the Object has a default implementation, meaning there would always be a function to chain up.

Reply:
Not only that, but also whether this function is mandatory. It might not have a default implementation but the caller might just call it always without checking if there's one, in which case it's mandatory

Reply:
Checking the code, it looks mandatory so unwrapping/expect here is OK.

Reply:
I am curious though if this should return one of the two glib enums we have that internally maps to a boolean type

Reply:
Return value should be a `glib::Propagation` I think

Reply:
@sdroege Thanks for checking and saving me the trip to C 🙏

I'll change the return value to `Propagation`